### PR TITLE
Fetch Cart: add bolt hints to customer-data cart

### PR DESCRIPTION
**Pull request by garmashdev**
# Description
Please include a summary of the change and which issue is fixed. Include the motivation for the changes, and comment on your PR if necessary for clarity.

Fixes: (link ticket)

#changelog Fetch Cart: add bolt hints to customer-data cart

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

**Comment by garmashdev** [comment]
Current dependencies on/for this PR:
* master
  * **PR #1660** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1660" target="_blank">Graphite</a> 
    * **PR #1672** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1672" target="_blank">Graphite</a>  👈
      * **PR #1667** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1667" target="_blank">Graphite</a> 
        * **PR #1671** <a href="https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1671" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/BoltApp/bolt-magento2/1672?utm_source=stack-comment).

File: Plugin/Magento/Checkout/CustomerData/Cart.php [hunk 1]
```diff
@@ -75,9 +85,12 @@ public function __construct(
     public function afterGetSectionData(CustomerDataCart $subject, array $result)
     {
         $quote = $this->getQuote();
+        $result['quoteMaskedId'] = null;
+        $result['boltCartHints'] = null;
         if ($quote->getId()) {
             try {
                 $result['quoteMaskedId'] = $this->getQuoteMaskedId((int)$quote->getId());
+                $result['boltCartHints'] = $this->boltHelperCart->getHints($quote->getId(), self::HINTS_TYPE);
```
Comment (vitaliyreznikov) [inline Plugin/Magento/Checkout/CustomerData/Cart.php:93]: any reason why we convert quote id to int in previous line but do not in this line?

**Review by vitaliyreznikov — APPROVED** [review]
LGTM, one small question.

File: Plugin/Magento/Checkout/CustomerData/Cart.php (hunk 1 above)
Comment (garmashdev) [reply Plugin/Magento/Checkout/CustomerData/Cart.php:93]: in method (i've created) getQuoteMaskedId i'm using strict declaration (getQuoteMaskedId(int $quoteId)), and for boltHelperCart->getHints(...) methods we don't have strict parameters.

**Pull request merged**